src/main/kotlin/graphics/scenery/CatmullRomSpline.kt

```diff
@@ -14,24 +15,27 @@ import kotlin.math.pow
  * [alpha] determines the kind of Catmull Rom Spline, set in range of 0..1, the
  * resulting curve for alpha = 0 is a standart Catmull Rom Spline, for alpha = 1 we get
  * a chordal Catmull Rom Spline.
+ *
+ * @author  Justin Buerger <burger@mpi-cbg.de>
  */
-class CatmullRomSpline(protected val controlPoints: List<Vector3f>, val n: Int = 100, val alpha: Float = 0.5f): Spline {
+class CatmullRomSpline(private val controlPoints: List<Vector3f>, val n: Int = 100, val alpha: Float = 0.5f,
```

Comment:
Have a close look here regarding what you want to have accessible and not. With the current declarations, controlPoints can't even be accessed from a derived class, while n and alpha are publicly accessible and can be changed!